dart/lib/src/platform_checker.dart

```diff
@@ -17,4 +22,23 @@ class PlatformChecker {
   bool isProfileMode() {
     return const bool.fromEnvironment('dart.vm.profile', defaultValue: false);
   }
+
+  final bool isWeb;
+
+  /// Indicates wether a native integration is available.
+  bool get hasNativeIntegration {
+    if (isWeb) {
+      // On web platform OS checks return the OS the browser is running on
+      return false;
+    }
+    if (platform.isAndroid || platform.isIOS || platform.isMacOS) {
+      return true;
+    }
+    return false;
```

Comment:
this `if` can be simplified, returns false if its web or if no condition is met

Reply:
`platform.isAndroid` returns true on web if it is a browser on an Android device. So no we can't simplify this.
I should probably a comment clarifying this.

Reply:
yeah that makes sense, lets add a comment then